core/src/main/scala/org/apache/spark/deploy/master/Master.scala

```diff
@@ -702,8 +712,6 @@ private[deploy] class Master(
     worker.addExecutor(exec)
     worker.endpoint.send(LaunchExecutor(masterUrl,
       exec.application.id, exec.id, exec.application.desc, exec.cores, exec.memory))
-    exec.application.driver.send(ExecutorAdded(
-      exec.id, worker.id, worker.hostPort, exec.cores, exec.memory))
```

Comment:
I think I didn't remove this one, just move to the above code in `ExecutorStateChanged `.